askama_shared/src/lib.rs

```diff
@@ -257,19 +258,71 @@ where
     vals.iter().map(|s| s.to_string()).collect()
 }
 
+/// Whitespace handling of the input source.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum Strip {
+    /// Don't strip any spaces in the input.
+    None,
+    /// Remove a single single newline at the end of the input. This is the default.
+    Tail,
+    /// Remove all whitespaces at the front and back all lines, and remove empty lines.
+    Trim,
+    /// Like Trim, but also replace runs of whitespaces with a single space.
+    Eager,
+}
+
+impl Default for Strip {
+    fn default() -> Self {
+        Strip::Tail
+    }
+}
+
+impl Strip {
+    pub fn apply<'a, S: Into<Cow<'a, str>>>(self, src: S) -> String {
+        let src = S::into(src);
```

Comment:
I think I'd prefer spelling this `src.into()`.

Reply:
You're right. I don't know why I used the more complicated syntax before. :)